app/lib/report_helper.rb

```diff
@@ -427,10 +421,21 @@ def top_taxons_details(samples, background_id, num_results, sort_by_key, only_sp
       taxon_counts = res["taxon_counts"]
       sample_id = pr.sample_id
       tax_2d = convert_2d(taxon_counts)
+      cleanup_genus_ids!(tax_2d)
+      validate_names!(tax_2d)
+      cleanup_missing_genus_counts!(tax_2d)
```

Comment:
Is this any different from cleanup_all?    This is good for now, but should be deduped further.